msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
Actually, I see that we have `tree.leaves(u)`, so we could do something like:

```
colour_nodes = set([1,2,5,6,8,9])
branch_colours = {u: "red" if set(t.leaves(u)) <= colour_nodes for u in t.nodes())}
```

although I imagine that for large trees this might be unacceptably slow, since it re-calculates the leaf sets for each node.